make-coe/make-coe.go

```diff
@@ -117,7 +138,7 @@ func (carina *MakeCOE) ListClusterTemplates() ([]common.ClusterTemplate, error)
 
 	results, err := carina.listClusterTypes()
 	if err != nil {
-		return nil, err
+		return nil, handleLibcarinaError(err)
```

Comment:
Since `carina.listClusterTypes` is in the cli, and we'd like to have the error wrapping happen as close as possible to the libcarina call, let's move this into `carina.listClusterTypes`.